nncf/utils.py

```diff
@@ -393,34 +394,42 @@ def should_consider_scope(scope_str: str, target_scopes: List[str], ignored_scop
                and not in_scope_list(scope_str, ignored_scopes)
 
 
-def save_module_training_state(module: torch.nn.Module, saved_state: Dict[torch.nn.Module, bool]) -> None:
-    for ch in module.children():
-        saved_state[ch] = ch.training
-        save_module_training_state(ch, saved_state)
+def save_module_state(module: Module) -> NamedTuple:
+    State = namedtuple('State', ['training_state', 'requires_grad_state'])
```

Comment:
Set up a proper class instead of using namedtuples.

Reply:
I've added a `_ModuleState` class to store necessary model fields. Please, take a look